### PR TITLE
hover refactor

### PR DESCRIPTION
**Pull request by dimacodota**

### CODE REVIEW

File: src/hovers/decorationState.ts [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+} from "vscode";
+import { Hover } from "../binary/requests/hovers";
+import { getLogoPath } from "../consts";
+
+const decorationType = window.createTextEditorDecorationType({
+  after: { margin: "0 0 0 1rem" },
+});
+
+let decoration: DecorationOptions | null;
+let decorationsDebounce: NodeJS.Timeout;
```
Comment (boaz-codota) [inline src/hovers/decorationState.ts:18]: the types are incorrect. Both are `undefined`, but both are not defined as undefined. Are you using vscode to lint the typescript?

File: src/hovers/hoverActionsHandler.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import { commands, Disposable, ExtensionContext } from "vscode";
+import { Hover, sendHoverAction } from "../binary/requests/hovers";
+
+let hoverActionsDisposable: Disposable[];
```
Comment (boaz-codota) [inline src/hovers/hoverActionsHandler.ts:4]: type should be `| undefined`

Comment (boaz-codota) [reply src/hovers/hoverActionsHandler.ts:4]: or you could initialize it with `[]`

File: src/consts.ts [hunk 3]
```diff
@@ -111,6 +111,14 @@ export function restartBackoff(attempt: number): number {
     MAX_SLEEP_TIME_BETWEEN_ATTEMPTS
   );
 }
+export function getLogoPath(context: ExtensionContext) : string {
+  return Uri.file(
+    path.join(
+      context.extensionPath,
+      LOGO_BY_THEME[window.activeColorTheme.kind]
```
Comment (boaz-codota) [inline src/consts.ts:118]: NOICE! (aka cool cool cool)

File: src/hovers/hoverHandler.ts [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+let currentHover: Hover | null | undefined = null;
+
+languages.registerHoverProvider(
+  { pattern: "**" },
+  {
+    provideHover(_document, position) {
+      return handleHoverShown(position);
+    },
+  }
+);
```
Comment (boaz-codota) [inline src/hovers/hoverHandler.ts:17]: why here and not in `extension.ts:activate()`?

File: src/hovers/hoverHandler.ts [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+    void setState({
+      [StatePayload.HOVER_SHOWN]: {
+        id: currentHover.id,
+        text: currentHover.message,
+        notification_type: currentHover.notification_type,
+        state: currentHover.state
+      },
+    });
+  }
+  return null;
```
Comment (boaz-codota) [inline src/hovers/hoverHandler.ts:30]: why?

Comment (boaz-codota) [reply src/hovers/hoverHandler.ts:30]: I think you can just delete this row

**Review by boaz-codota — APPROVED** [review]
Cool, had somethink I'd love you to address. Overall LGTM.

**Comment by boaz-codota** [comment]
Also note that the build is failing

File: src/hovers/hoverHandler.ts (hunk 5 above)
Comment (dimacodota) [reply src/hovers/hoverHandler.ts:30]: i changed it a bit, but the `provideHover`  expects to  `vscode.ProviderResult` as a result
https://code.visualstudio.com/api/references/vscode-api#HoverProvider

**Pull request merged**